File: java/vtgate-client/src/main/java/com/youtube/vitess/vtgate/QueryResponse.java
package com.youtube.vitess.vtgate;

public class QueryResponse {
	private QueryResult result;
	private Object session;
	private String error;

	public QueryResponse(QueryResult result, Object session, String error) {
		this.result = result;
		this.session = session;
		this.error = error;
	}

	public QueryResult getResult() {
		return result;
	}

	public Object getSession() {
		return session;
	}

	public String getError() {
		return error;
	}
}

File: go/sqltypes/query_response.go
package sqltypes

import (
	"vitess.io/vitess/go/vt/vterrors"
)

// QueryResponse represents a query response for ExecuteBatch.
type QueryResponse struct {
	QueryResult *Result
	QueryError  error
}

// QueryResponsesEqual compares two arrays of QueryResponse.
// They contain protos, so we cannot use reflect.DeepEqual.
func QueryResponsesEqual(r1, r2 []QueryResponse) bool {
	if len(r1) != len(r2) {
		return false
	}
	for i, r := range r1 {
		if !r.QueryResult.Equal(r2[i].QueryResult) {
			return false
		}
		if !vterrors.Equals(r.QueryError, r2[i].QueryError) {
			return false
		}
	}
	return true
}
